tests/demos/test_demos_run.py

```diff
@@ -57,4 +57,16 @@ def py_file(rst_file, tmpdir, monkeypatch):
 
 @pytest.mark.skipcomplex  # Will need to add a seperate case for a complex demo.
 def test_demo_runs(py_file, env):
+    if basename(py_file) == "qgbasinmodes.py":
+        try:
+            from slepc4py import SLEPc  # noqa: F401
+        except ImportError:
+            pytest.skip(msg="SLEPc unavailable, skipping qgbasinmodes.py")
+
+    if basename(py_file) in ("DG_advection.py", "qgbasinmodes.py"):
+        try:
+            import matplotlib  # noqa: F401
+        except ImportError:
+            pytest.skip(msg="Matplotlib unavailable, skipping DG_advection.py")
```

Comment:
Wrong error message.

```suggestion
            pytest.skip(msg=f"Matplotlib unavailable, skipping {basename(py_file)}")
```